src/auth/src/credentials/service_account_credential.rs

```diff
@@ -48,6 +48,18 @@ pub(crate) struct ServiceAccountInfo {
     universe_domain: String,
 }
 
+impl std::fmt::Debug for ServiceAccountInfo {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("ServiceAccountInfo")
+            .field("client_email", &self.client_email)
+            .field("private_key_id", &"[censored]")
+            .field("private_key", &"[censored]")
```

Comment:
For a future improvement: maybe printing the first X digits or a SHA would be useful for troubleshooting.

Reply:
> first X digits

In this case, it will be the BEGIN PRIVATE KEY nonsense

> SHA would be useful

I could see that. #872